etl/steps/data/garden/democracy/2024-03-07/bmr.py

```diff
@@ -0,0 +1,770 @@
+from owid.catalog.tables import Table, concat
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+# IMPUTE: We will infer indicator values for some countries based on their historical equivalences.
+path = paths.directory / "bmr.countries_impute.yml"
+COUNTRIES_IMPUTE = yaml.safe_load(path.read_text())
+
+# Overlapping countries as expected when counting countries
+COUNTRIES_OVERLAP = json.loads((paths.directory / "bmr.countries_overlap.json").read_text())
+COUNTRIES_OVERLAP = [{int(k): set(v) for k, v in overlaps.items()} for overlaps in COUNTRIES_OVERLAP]
+
+# REGION AGGREGATES
+REGIONS = {
+    "Africa": {},
+    "Asia": {},
+    "North America": {},
+    "South America": {},
+    "Europe": {},
+    "Oceania": {},
+}
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("bmr")
+    ds_regions = paths.load_dataset("regions")
+    ds_population = paths.load_dataset("population")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["bmr"].reset_index()
+
+    #
+    # Process data.
+    #
+    # Harmonize country names
+    tb = geo.harmonize_countries(df=tb, countries_file=paths.country_mapping_path)
+
+    # Identify observations from countries that would overlap with other countries (Russia with USSR in 1991)
+    tb = remove_country_overlaps(tb)
+
+    # Keep relevant columns and rename
+    tb = tb[["country", "year", "democracy_omitteddata", "democracy_femalesuffrage"]]
+    tb = tb.rename(
+        columns={
+            "democracy_omitteddata": "regime",
+            "democracy_femalesuffrage": "regime_womsuffr",
+        }
+    )
+
+    # Drop all-NaN rows
+    cols_subset = [col for col in tb.columns if col not in ["country", "year"]]
+    tb = tb.dropna(subset=cols_subset, how="all")
+
+    # Impute values for missing countries
+    # That is: Estimate whether a country was a democracy or not, based on its historical equivalents.
+    tb = add_imputes(tb)
+
+    # Refine
+    ## Set NaNs to `regime_womsuffr` whenever `regime` is NaN. Discussion on this here: https://github.com/owid/owid-issues/issues/1421#issuecomment-1992057534
+    tb.loc[tb["regime"].isna(), "regime_womsuffr"] = pd.NA
+
+    ## Add democracy age / experience
+    tb = tb.sort_values(["country", "year"])
+
+    # Add years in democracy (consecutive and total)
+    tb = add_years_in_democracy(tb)
+
+    # Add categories for "years in democracy"
+    tb = add_categories_years_in_democracy(tb)
+
+    # Get number of countries in democracy, by region and World
+    tb_num_countries, tb_num_countries_years_consec = make_tables_country_counters(tb, ds_regions)
+
+    # Get number of people in democracy, by region and World
+    tb_pop, tb_pop_years_consec = make_tables_population_counters(tb, ds_regions, ds_population)
```

Comment:
Are you expecting NAs in the `population_regime_ws` column in `tb_pop`? I get some when I run it interactively e.g. for World in 2020

Reply:
It is expected, since I was trying to not import this indicator to Grapher. Hence I assigned it to NaN for all years and regions when category was `-1`. However, I think it makes more sense to keep it with the actual value, and do this NaN-assignment in the Grapher step!